app/models/habit_builder.rb

```diff
@@ -0,0 +1,32 @@
+class HabitBuilder
+
+  attr_reader :user, :params
+  def initialize(user, params)
+    @user, @params = user, params
+  end
+
+  def build
+    if daily_habit?
+      habit = Habit.new(name: params[:name], frequency: "daily")
+    else
+      habit = Habit.new(name: params[:name], frequency: habit_frequency)
+    end
+
+    habit.user = user
+    habit.save!
+  end
+
+  private
+
+  def habit_frequency
+    days = params[:frequency].find_all do |k, v|
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Prefer `select` over `find_all`.